cancatlib/__init__.py

```diff
@@ -680,6 +680,10 @@ def _isotp_get_msg(self, rx_arbid, start_index=0, service=None, timeout=None):
         complete = False
         starttime = lasttime = time.time()
 
+        if service is not None and isinstance(service, int):
```

Comment:
if `isinstance(service, int)` then the first clause must also be true, thus superfluous.  meanwhile, if it's an int, it must be <256 unless you want to force it to be the lowest 8 bits

```suggestion
        if isinstance(service, int):
```

and further, i believe this is an attempt to overcome the "int because a single byte slice" problem, eg.  `foobytes[3]`
a better solution for this (since we're being programmatic here and can apply some expectations to our code) would be:
`foobytes[3:4]` which still grabs the 1 byte, but keeps it in a bytes() object.

does this make sense?  what are your thoughts?

Reply:
This makes sense and I think works fine.  The current check I do below this point is
```
if msg[:len(service)] == service or msg[0] == 0x7f: 
```
The reason I use `len(service)` is because this redesign of the `_isotp_get_msg` function is to allow UDS services to specify more than a single byte.  For example in ReadDID it now should check for the correct positive response SID + the DID being read.